spec/acceptance/orchestrator/with_callback_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+require 'test_doubles'
+
+describe LightService::Orchestrator do
+  class TestWithCallback
+    extend LightService::Orchestrator
+
+    def self.run(context = {})
+      with(context).reduce(steps)
+    end
+
+    def self.steps
+      [
+        SetUpContextAction,
+        with_callback(IterateCollectionAction,
```

Comment:
Hi! It's been a while since I made some feedback to this project. We're glad to use light-service again in a completely new project that does some ETL work as well. 🥂 🎆 

With that said, It took me a good 10-20 minutes to re-read the spec and source code and I still don't understand it. It helps that you're not here to explain it as to test whether it's readable on its own or not.

I thought that the callback was `InteractionColllectionAction` and not `[IncrementCountAction, AddToTotalAction]` (or did I get this incorrectly?). So many things are at play here: 

1. Maybe callbacks should have their own naming convention akin to naming actions as `*Action`, since the example might be too simple. Or at the very least, make it explicit which classes are the callbacks.

2. Reading `def with_callback(action, steps)` alone in `orchestrator.rb`, it seems to suggest, by proximity, that `action` is the callback, but in this example, `IterationCollectionAction` seems to be doing the main thing (to iterate over numbers) and the counters seems to be doing the side tasks (updating counters). Also `IterationCollectionAction` ... `expects :callback`, which tells me that this is not the callback itself.

Maybe if the callback API is something like this:

```ruby
reduce(
  IterateCollectionAction,
  callback: [IncrementCountAction, AddToTotalAction]
)
```
... it would make it more explicit which are the callbacks.